Dockerfile.builder

```diff
@@ -0,0 +1,28 @@
+# syntax=docker/dockerfile:1
+
+ARG CUDA=12.2
+FROM sameli/manylinux2014_x86_64_cuda_${CUDA} AS build
```

Comment:
Do we want to rely on someone else's dockerfile here or am I being unduly paranoid? All they do is install CUDA from rpm files as I remember, we could just copy-paste

Reply:
It seems to be regularly updated at the moment so no harm in using it and it means avoiding having to deal with horrible version strings in the CUDA filenames.